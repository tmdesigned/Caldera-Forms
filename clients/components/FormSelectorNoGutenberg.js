import React from 'react';
import PropTypes from 'prop-types';

import { FormGroup,HelpBlock } from 'react-bootstrap';

/**
 * Form selector with no dependency on Gutenberg
 *
 * @since 1.7.0
 *
 * @param props
 * @returns {*}
 * @constructor
 */
export const FormSelectorNoGutenberg = (props) => {
    const idAttr = props.idAttr ? props.idAttr :'caldera-forms-form-chooser';
    return (
        <FormGroup controlId={idAttr}>
            <label
                htmlFor={idAttr}
            >
                Choose Form
            </label>
            <select
                id={idAttr}
                value={props.selected || ''}
                className="select"
                onChange={(e) => {
                    props.onChange(e.target.value)
                }}
            >
                <option value=''></option>
                {Object.keys(props.forms).map( (formsIndex) => {
                    const form = props.forms[formsIndex];
                    return (
                        <option
                            key={form.ID}
                            value={form.ID}
                        >
                            {form.name}
                        </option>
                    )
                })}
            </select>
            <HelpBlock>Choose a form to edit privacy settings for.</HelpBlock>
        </FormGroup>
    )
};

FormSelectorNoGutenberg.propTypes = {
    forms: PropTypes.array.isRequired,
    onChange: PropTypes.func.isRequired,
    selected: PropTypes.string,
    idAttr: PropTypes.string
};