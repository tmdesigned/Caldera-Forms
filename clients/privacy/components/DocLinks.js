import React from 'react';
import { Twemoji } from 'react-emoji-render';
import {ListGroup, ListGroupItem } from 'react-bootstrap'
import PropTypes from 'prop-types';

/**
 * Shows the doc links in the Privacy settings page
 *
 * @since 1.7.2
 *
 * @param props
 * @return {*}
 */
export const DocLinks = (props) => {
    return (
        <React.Fragment>
            <ListGroup
                style={{padding:'12px'}}
            >
                <ListGroupItem>
                    <Twemoji text=":eyes:" />
                    <a href="https://calderaforms.com/doc/setting-caldera-forms-gdpr-data-requests/?utm_source=wp-admin&utm_campaign=privacy-settings">
                        Setting Up Caldera Forms For GDPR Requests
                    </a>
                </ListGroupItem>
                <ListGroupItem>
                    <Twemoji text=":eyes:" />
                    <a href="https://calderaforms.com/gdpr/?utm_source=wp-admin&utm_campaign=privacy-settings">
                       All Caldera Forms GDPR Compliance Tools
                    </a>
                </ListGroupItem>
				<ListGroupItem>
                    {props.children}
                </ListGroupItem>
            </ListGroup>
        </React.Fragment>
    )
};

DocLinks.propTypes = {
	extraLink: PropTypes.string,
	extraLinkText: PropTypes.string
};