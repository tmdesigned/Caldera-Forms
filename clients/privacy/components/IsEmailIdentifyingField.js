import React from  'react';
import PropTypes from 'prop-types';
import { FormGroup,FormControl,ControlLabel,HelpBlock,Panel,PanelGroup,Checkbox } from 'react-bootstrap';

/**
 * Determine if field is a email identifying field
 *
 * @since 1.7.0
 *
 * @param {Object} field
 * @param {Object} privacySettings
 * @returns {*}
 */
function fieldIsEmailIdentifying(field,privacySettings ){
    return privacySettings.emailIdentifyingFields.length && privacySettings.emailIdentifyingFields.includes(field.ID);
}

/**
 * Setting for a form being an email identifying field
 *
 * @since 1.7.0
 *
 * @param props
 * @returns {*}
 * @constructor
 */
export const IsEmailIdentifyingField = (props)  => {
    const idAttr  = `caldera-forms-privacy-gdpr-is-email-identifiying-${props.field.ID}`;
    if( 'email' === props.field.type || 'text' === props.field.text ){
        return (
            <FormGroup>
                <ControlLabel
                    htmlFor={idAttr}
                >
                    Email Identifying Field?
                </ControlLabel>
                <Checkbox
                    id={idAttr}
                    onChange={() => {
                            props.onCheck(props.field.ID)
                        }
                    }
                    checked={fieldIsEmailIdentifying(props.field,props.privacySettings)}
                >
                    <span style={{
                        marginLeft: '12px'
                    }}>
                        Yes
                    </span>
                </Checkbox>
                <HelpBlock
                    className={'screen-reader-text'}
                >
                    Can this field be used to determine whose data an entry belongs to?
                </HelpBlock>
            </FormGroup>
        )
    }
    return (
        <div
            id={idAttr}
        >
            Not an email or text field
        </div>
    )

};

IsEmailIdentifyingField.propTypes=  {
    field: PropTypes.object.isRequired,
    privacySettings: PropTypes.object.isRequired,
    onCheck: PropTypes.func.isRequired,
};