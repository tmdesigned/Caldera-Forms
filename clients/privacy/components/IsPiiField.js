import React from  'react';
import PropTypes from 'prop-types';
import { FormGroup,ControlLabel,HelpBlock,Checkbox } from 'react-bootstrap';

/**
 * Determine if field is a PII field
 *
 * @since 1.7.0
 *
 * @param {Object} field
 * @param {Object} privacySettings
 * @returns {*}
 */
function fieldIsPii(field,privacySettings ) {
    return privacySettings.piiFields.length && privacySettings.piiFields.includes(field.ID);
}

/**
 * Setting to determine if this is a PII field
 *
 * @since 1.7.0
 *
 * @param props
 * @returns {*}
 * @constructor
 */
export const IsPiiField = (props) => {
    const idAttr  = `caldera-forms-privacy-gdpr-is-pii-field-${props.field.ID}`;
    return (
        <FormGroup>
            <ControlLabel
               htmlFor={idAttr}
            >
               Personally Identifying Field?
            </ControlLabel>
            <Checkbox
                id={idAttr}
                onChange={() => {
                        props.onCheck(props.field.ID)
                    }
                }
                checked={fieldIsPii(props.field,props.privacySettings)}
            >
                <span style={{
                    marginLeft: '12px'
                }}>
                    Yes
                </span>
            </Checkbox>
            <HelpBlock
                className={'screen-reader-text'}
            >
                Does field contain personally identifying data?
            </HelpBlock>
        </FormGroup>
    );

};

IsPiiField.propTypes = {
    field: PropTypes.object.isRequired,
    privacySettings: PropTypes.object.isRequired,
    onCheck: PropTypes.func.isRequired,
};