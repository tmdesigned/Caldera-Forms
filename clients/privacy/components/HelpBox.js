import React from 'react';
import PropTypes from 'prop-types';
import {Twemoji} from 'react-emoji-render';
import {DocLinks} from "./DocLinks";
import {ListGroup, ListGroupItem } from 'react-bootstrap'
export const HelpBox = (props) => {
	return (
		<div
			style={{
				backgroundColor: '#fff'
			}}
			className={'col-xs-4'}
		>
			{props.saveButton &&
				<p
					style={{
						margin: 0,
						padding: '.7em 1em',
						borderBottom: '1px solid #eee'
					}}
				>
					{props.saveButton}
				</p>

			}
			<h3>
				<Twemoji text=":volcano: Terms" />
			</h3>
			<ListGroup
				style={{padding:'12px'}}
			>
				<ListGroupItem>
					<Twemoji text=":eyes:"/>Email Identifying Fields: The field(s) of your form that can be used to determine whose data an entry belongs to.
				</ListGroupItem>
				<ListGroupItem>
					<Twemoji text=":eyes:"/>Personally Identifying Fields (PII): The field(s) of your form that contain PII about the person identified in the email identifying field.
				</ListGroupItem>
			</ListGroup>

			<h3>
				<Twemoji text=":volcano: Documentation" />
			</h3>
			<DocLinks/>
		</div>
	);
};

HelpBox.propTypes = {
	saveButton: PropTypes.oneOfType([
		PropTypes.node,
		PropTypes.object
	]),
};