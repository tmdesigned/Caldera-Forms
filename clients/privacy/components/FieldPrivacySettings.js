import React from 'react';
import PropTypes from 'prop-types';
import {FormGroup, FormControl, ControlLabel, HelpBlock, Panel, PanelGroup, Checkbox} from 'react-bootstrap';
import {IsEmailIdentifyingField} from "./IsEmailIdentifyingField";
import {IsPiiField} from "./IsPiiField";

/**
 * Privacy Settings for one field of a form
 *
 * @since 1.7.0
 *
 * @param props
 * @returns {*}
 * @constructor
 */
export const FieldPrivacySettings = (props) => {
    return (
        <section
            style={{
                backgroundColor: '#fff',
                margin: '1em',
            }}
            className={'row'}
        >
            <h4
                style={{
                    margin: 0,
                    padding: '.7em 1em',
                    borderBottom: '1px solid #eee'
                }}
            >
                {props.field.name}
            </h4>
            <div
                className={'col-xs-6'}
                style={{
                    padding: '12px'
                }}
            >
                <IsPiiField
                    field={props.field}
                    privacySettings={props.privacySettings}
                    onCheck={props.onCheckIsPii}
                />
            </div>

            <div
                className={'col-xs-6'}
                style={{
                    padding: '12px'
                }}
            >
                <IsEmailIdentifyingField
                    className={'col-xs-6'}
                    field={props.field}
                    privacySettings={props.privacySettings}
                    onCheck={props.onCheckIsEmail}
                />
            </div>


        </section>
    );
};

FieldPrivacySettings.propTypes = {
    field: PropTypes.object.isRequired,
    formId: PropTypes.string.isRequired,
    privacySettings: PropTypes.object.isRequired,
    onCheckIsEmail: PropTypes.func.isRequired,
    onCheckIsPii: PropTypes.func.isRequired
};