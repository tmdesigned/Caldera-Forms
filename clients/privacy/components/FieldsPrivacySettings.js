import React from 'react';
import PropTypes from "prop-types";
import {FieldPrivacySettings} from "./FieldPrivacySettings";

/**
 * All field-related privacy settings for a form
 *
 * @since 1.7.0
 *
 * @param props
 * @returns {*}
 * @constructor
 */
export const FieldsPrivacySettings = (props) => {
    return (
        <div>
            {Object.keys(props.fields).map( (fieldId ) => {
                const field = props.fields[fieldId];
                if ( field && field.hasOwnProperty( 'ID' ) ) {
                    const idAttr = `caldera-forms-privacy-field-settings-${field.ID}`;
                    return (
                        <FieldPrivacySettings
                            id={idAttr}
                            key={idAttr}
                            field={field}
                            formId={props.formId}
                            privacySettings={props.privacySettings}
                            onCheckIsEmail={props.onCheckIsEmail}
                            onCheckIsPii={props.onCheckIsPii}
                        />
                    );
                } else {
                    <p>Invalid Field</p>
                }

            })}
        </div>
    )
};

FieldsPrivacySettings.propTypes = {
    fields: PropTypes.object.isRequired,
    formId: PropTypes.string.isRequired,
    privacySettings: PropTypes.object.isRequired,
    onCheckIsEmail: PropTypes.func.isRequired,
    onCheckIsPii: PropTypes.func.isRequired
};