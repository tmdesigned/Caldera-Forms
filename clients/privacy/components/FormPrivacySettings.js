import React from 'react';
import PropTypes from 'prop-types';
import {FormGroup, ControlLabel, Checkbox, HelpBlock, Button } from 'react-bootstrap';
import {FieldsPrivacySettings} from "./FieldsPrivacySettings";
import remove from 'lodash.remove';
import {HelpBox} from "./HelpBox";

/**
 * All privacy settings for a form
 *
 * @since 1.7.0
 *
 * @param props
 * @returns {*}
 * @constructor
 */
export const FormPrivacySettings = (props) => {
    /**
     * Toggle exporter enabled
     *
     * @since 1.7.0
     */
    const toggleEnable = () => {
        props.onStateChange({
            ...props.privacySettings,
            privacyExporterEnabled: !props.privacySettings.privacyExporterEnabled,
        });
    };

    /**
     * Add a field to list if in list, remove if not in list
     *
     * @since 1.7.0
     * @param {Array} fields
     * @param {String} fieldId
     * @returns {*}
     */
    function updateFieldsList(fields, fieldId) {
        if (!fields.length || !fields.includes(fieldId)) {
            fields.push(fieldId)
        } else {
            if( 1 === fields.length ){
                fields = [];
            }else{
                fields = remove(fields, (ID) => {
                    return ID === fieldId;
                });
            }

        }
        return fields;
    }

    /**
     * Toggle a field as an email field to identify users by or not.
     *
     * @since 1.7.0
     *
     * @param {String} fieldId
     */
    const toggleIsEmail = (fieldId) => {
        let emailIdentifyingFields = updateFieldsList(props.privacySettings.emailIdentifyingFields, fieldId);
        props.onStateChange({
            ...props.privacySettings,
            emailIdentifyingFields,
        });
    };

    /**
     * Toggle a field as a PII field or not
     *
     * @since 1.7.0
     *
     * @param fieldId
     */
    const toggleIsPii = (fieldId) => {
        let piiFields = updateFieldsList(props.privacySettings.piiFields, fieldId);
        props.onStateChange({
            ...props.privacySettings,
            piiFields,
        });
    };


    return (
        <section
            className={'layout-grid'}
        >
            <h2>{props.form.name}</h2>

            <div
                className={'row'}
            >
                <div
                    className={'col-xs-8'}
                >
                    <FormGroup controlId={`caldera-forms-privacy-gdpr-enable-${props.form.ID}`}>
                        <ControlLabel>
                            Enable GDPR Exporter For Form
                        </ControlLabel>
                        <Checkbox
                            onChange={toggleEnable}
                            checked={props.privacySettings.privacyExporterEnabled}
                        >
                            Enable
                        </Checkbox>
                        <HelpBlock>If checked data for this form will be added to GDPR personal data requests, and deletes.</HelpBlock>
                    </FormGroup>

                    {props.privacySettings.privacyExporterEnabled &&
                        <section>
                            <h3>Field Settings</h3>
                            <FieldsPrivacySettings
                                fields={props.form.fields}
                                formId={props.form.ID}
                                privacySettings={props.privacySettings}
                                onCheckIsEmail={toggleIsEmail}
                                onCheckIsPii={toggleIsPii}
                            />
                        </section>
                    }
                </div>
                <HelpBox
                    saveButton={
						<button
							type="submit"
							onClick={() => {
								props.onSave(props.privacySettings, props.form.ID)
							}}
							className={'primary button button-primary'}
						>
							Save
						</button>
                    }
                />
            </div>




        </section>
    );


};

FormPrivacySettings.propTypes = {
    form: PropTypes.object.isRequired,
    onSave: PropTypes.func.isRequired,
    privacySettings: PropTypes.shape({
        emailIdentifyingFields: PropTypes.array,
        piiFields: PropTypes.array,
        privacyExporterEnabled: PropTypes.bool,
        fields: PropTypes.object,
    }).isRequired,
    onStateChange: PropTypes.func.isRequired,
};