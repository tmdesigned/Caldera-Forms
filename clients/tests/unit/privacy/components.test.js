import renderer from 'react-test-renderer';
import {FieldGroup} from "../../../privacy/components/FieldGroup";
import {HelpBox} from "../../../privacy/components/HelpBox";
import {DocLinks} from "../../../privacy/components/DocLinks";
import {IsEmailIdentifyingField} from "../../../privacy/components/IsEmailIdentifyingField";
import {FormPrivacySettings} from "../../../privacy/components/FormPrivacySettings";
import {FieldsPrivacySettings} from "../../../privacy/components/FieldsPrivacySettings";
import {FieldPrivacySettings} from "../../../privacy/components/FieldPrivacySettings";
import {IsPiiField} from "../../../privacy/components/IsPiiField";

describe( 'Field group component', () => {
	it('Works with help text',() => {
		const fieldGroup = renderer.create(
			<FieldGroup
				id={'control-22'}
				label={'Who'}
				help={'Who to say hi to'}
			/>
		);
		expect( fieldGroup.toJSON() ).toMatchSnapshot();
	});

	it('Help text is optional',() => {
		const fieldGroup = renderer.create(
			<FieldGroup
				id={'control-22'}
				label={'Who'}
			/>
		);
		expect( fieldGroup.toJSON() ).toMatchSnapshot();
	});
});

describe( 'Helpbox', () => {
	it( 'Does what it does. This needs broken up', () => {
		const helpBox = renderer.create(
			<HelpBox
				saveButton={<button>Push Me</button>}
			/>
		);
		expect( helpBox.toJSON() ).toMatchSnapshot();

	});
});

describe( 'Docs links', () => {
	it( 'Renders', () => {
		const helpBox = renderer.create(
			<DocLinks/>
		);
		expect( helpBox.toJSON() ).toMatchSnapshot();

	});
});

describe( 'Settings', () => {
	const field = {'ID': 'fld_9899154', 'name': 'Priceaa', 'type': 'hidden'};
	const fields = {
		'fld_9899154': field,
		'fld_4917648': {'ID': 'fld_4917648', 'name': 'Email', 'type': 'email'},
		'fld_1081036': {'ID': 'fld_1081036', 'name': 'Total', 'type': 'calculation'}
	};
	const formId = 'CF5b197831b60ae';
	const settings = {
		'ID': formId,
		'name': 'Contact Form',
		'fields': fields,
		'emailIdentifyingFields': ['fld_4917648'],
		'piiFields': ['fld_9899154', 'fld_4917648'],
		'privacyExporterEnabled': true
	};

	describe('Email and PII Identify Field settings', () => {
		describe('Email identifying field', () => {
			it('Works', () => {
				const emailSettings = renderer.create(
					<IsEmailIdentifyingField
						field={field}
						privacySettings={settings}
						onCheck={() => {
						}}
					/>
				);
				expect(emailSettings.toJSON()).toMatchSnapshot();
			});
		});
		describe('Pii field', () => {
			it('Works', () => {
				const piiField = renderer.create(
					<IsPiiField
						field={field}
						privacySettings={settings}
						onCheck={() => {}}
					/>
				);
				expect(piiField.toJSON()).toMatchSnapshot();

			});
		});

	});

	describe('Field settings ', () => {
		it('Works', () => {
			const fieldPrivacySettings = renderer.create(
				<FieldPrivacySettings
					formId={formId}
					field={field}
					privacySettings={settings}
					onCheck={() => {}}
					onCheckIsEmail={() => {}}
					onCheckIsPii={() => {}}
				/>
			);
			expect(fieldPrivacySettings.toJSON()).toMatchSnapshot();

		});
	});
	describe('Fields settings ', () => {
		it('Works', () => {
			const fieldsPrivacySettings = renderer.create(
				<FieldsPrivacySettings
					formId={formId}
					fields={fields}
					privacySettings={settings}
					onCheckIsEmail={() => {}}
					onCheckIsPii={() => {}}
				/>
			);
			expect(fieldsPrivacySettings.toJSON()).toMatchSnapshot();

		});
	});
	describe( 'Form privacy settings', () => {
		it( 'Works', () => {
			const formPrivacySettings = renderer.create(
				<FormPrivacySettings
					form={{
						ID: formId,
						fields: fields
					}}
					onSave={() => {}}
					privacySettings={settings}
					onStateChange={() => {}}
				/>
			)
		});
	});
});
