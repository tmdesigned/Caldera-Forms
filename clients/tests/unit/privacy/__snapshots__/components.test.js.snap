// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Docs links Renders 1`] = `
<ul
  className="list-group"
  style={
    Object {
      "padding": "12px",
    }
  }
>
  <li
    className="list-group-item"
  >
    <span
      className="undefined"
    >
      <img
        alt="ðŸ‘€"
        src="http://twemoji.maxcdn.com/2/72x72/1f440.png"
        style={
          Object {
            "height": "1em",
            "margin": "0 .05em 0 .1em",
            "verticalAlign": "-0.1em",
            "width": "1em",
          }
        }
      />
    </span>
    <a
      href="https://calderaforms.com/doc/setting-caldera-forms-gdpr-data-requests/?utm_source=wp-admin&utm_campaign=privacy-settings"
    >
      Setting Up Caldera Forms For GDPR Requests
    </a>
  </li>
  <li
    className="list-group-item"
  >
    <span
      className="undefined"
    >
      <img
        alt="ðŸ‘€"
        src="http://twemoji.maxcdn.com/2/72x72/1f440.png"
        style={
          Object {
            "height": "1em",
            "margin": "0 .05em 0 .1em",
            "verticalAlign": "-0.1em",
            "width": "1em",
          }
        }
      />
    </span>
    <a
      href="https://calderaforms.com/gdpr/?utm_source=wp-admin&utm_campaign=privacy-settings"
    >
      All Caldera Forms GDPR Compliance Tools
    </a>
  </li>
  <li
    className="list-group-item"
  />
</ul>
`;

exports[`Field group component Help text is optional 1`] = `
<div
  className="form-group"
>
  <label
    className="control-label"
    htmlFor="control-22"
  >
    Who
  </label>
  <input
    className="form-control"
    id="control-22"
  />
</div>
`;

exports[`Field group component Works with help text 1`] = `
<div
  className="form-group"
>
  <label
    className="control-label"
    htmlFor="control-22"
  >
    Who
  </label>
  <input
    className="form-control"
    id="control-22"
  />
  <span
    className="help-block"
  >
    Who to say hi to
  </span>
</div>
`;

exports[`Helpbox Does what it does. This needs broken up 1`] = `
<div
  className="col-xs-4"
  style={
    Object {
      "backgroundColor": "#fff",
    }
  }
>
  <p
    style={
      Object {
        "borderBottom": "1px solid #eee",
        "margin": 0,
        "padding": ".7em 1em",
      }
    }
  >
    <button>
      Push Me
    </button>
  </p>
  <h3>
    <span
      className=""
    >
      <img
        alt="ðŸŒ‹"
        src="http://twemoji.maxcdn.com/2/72x72/1f30b.png"
        style={
          Object {
            "height": "1em",
            "margin": "0 .05em 0 .1em",
            "verticalAlign": "-0.1em",
            "width": "1em",
          }
        }
      />
       Terms
    </span>
  </h3>
  <ul
    className="list-group"
    style={
      Object {
        "padding": "12px",
      }
    }
  >
    <li
      className="list-group-item"
    >
      <span
        className="undefined"
      >
        <img
          alt="ðŸ‘€"
          src="http://twemoji.maxcdn.com/2/72x72/1f440.png"
          style={
            Object {
              "height": "1em",
              "margin": "0 .05em 0 .1em",
              "verticalAlign": "-0.1em",
              "width": "1em",
            }
          }
        />
      </span>
      Email Identifying Fields: The field(s) of your form that can be used to determine whose data an entry belongs to.
    </li>
    <li
      className="list-group-item"
    >
      <span
        className="undefined"
      >
        <img
          alt="ðŸ‘€"
          src="http://twemoji.maxcdn.com/2/72x72/1f440.png"
          style={
            Object {
              "height": "1em",
              "margin": "0 .05em 0 .1em",
              "verticalAlign": "-0.1em",
              "width": "1em",
            }
          }
        />
      </span>
      Personally Identifying Fields (PII): The field(s) of your form that contain PII about the person identified in the email identifying field.
    </li>
  </ul>
  <h3>
    <span
      className=""
    >
      <img
        alt="ðŸŒ‹"
        src="http://twemoji.maxcdn.com/2/72x72/1f30b.png"
        style={
          Object {
            "height": "1em",
            "margin": "0 .05em 0 .1em",
            "verticalAlign": "-0.1em",
            "width": "1em",
          }
        }
      />
       Documentation
    </span>
  </h3>
  <ul
    className="list-group"
    style={
      Object {
        "padding": "12px",
      }
    }
  >
    <li
      className="list-group-item"
    >
      <span
        className="undefined"
      >
        <img
          alt="ðŸ‘€"
          src="http://twemoji.maxcdn.com/2/72x72/1f440.png"
          style={
            Object {
              "height": "1em",
              "margin": "0 .05em 0 .1em",
              "verticalAlign": "-0.1em",
              "width": "1em",
            }
          }
        />
      </span>
      <a
        href="https://calderaforms.com/doc/setting-caldera-forms-gdpr-data-requests/?utm_source=wp-admin&utm_campaign=privacy-settings"
      >
        Setting Up Caldera Forms For GDPR Requests
      </a>
    </li>
    <li
      className="list-group-item"
    >
      <span
        className="undefined"
      >
        <img
          alt="ðŸ‘€"
          src="http://twemoji.maxcdn.com/2/72x72/1f440.png"
          style={
            Object {
              "height": "1em",
              "margin": "0 .05em 0 .1em",
              "verticalAlign": "-0.1em",
              "width": "1em",
            }
          }
        />
      </span>
      <a
        href="https://calderaforms.com/gdpr/?utm_source=wp-admin&utm_campaign=privacy-settings"
      >
        All Caldera Forms GDPR Compliance Tools
      </a>
    </li>
    <li
      className="list-group-item"
    />
  </ul>
</div>
`;

exports[`Settings Email and PII Identify Field settings Email identifying field Works 1`] = `
<div
  id="caldera-forms-privacy-gdpr-is-email-identifiying-fld_9899154"
>
  Not an email or text field
</div>
`;

exports[`Settings Email and PII Identify Field settings Pii field Works 1`] = `
<div
  className="form-group"
>
  <label
    className="control-label"
    htmlFor="caldera-forms-privacy-gdpr-is-pii-field-fld_9899154"
  >
    Personally Identifying Field?
  </label>
  <div
    className="checkbox"
  >
    <label
      title=""
    >
      <input
        checked={true}
        disabled={false}
        id="caldera-forms-privacy-gdpr-is-pii-field-fld_9899154"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        style={
          Object {
            "marginLeft": "12px",
          }
        }
      >
        Yes
      </span>
    </label>
  </div>
  <span
    className="screen-reader-text help-block"
  >
    Does field contain personally identifying data?
  </span>
</div>
`;

exports[`Settings Field settings  Works 1`] = `
<section
  className="row"
  style={
    Object {
      "backgroundColor": "#fff",
      "margin": "1em",
    }
  }
>
  <h4
    style={
      Object {
        "borderBottom": "1px solid #eee",
        "margin": 0,
        "padding": ".7em 1em",
      }
    }
  >
    Priceaa
  </h4>
  <div
    className="col-xs-6"
    style={
      Object {
        "padding": "12px",
      }
    }
  >
    <div
      className="form-group"
    >
      <label
        className="control-label"
        htmlFor="caldera-forms-privacy-gdpr-is-pii-field-fld_9899154"
      >
        Personally Identifying Field?
      </label>
      <div
        className="checkbox"
      >
        <label
          title=""
        >
          <input
            checked={true}
            disabled={false}
            id="caldera-forms-privacy-gdpr-is-pii-field-fld_9899154"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            style={
              Object {
                "marginLeft": "12px",
              }
            }
          >
            Yes
          </span>
        </label>
      </div>
      <span
        className="screen-reader-text help-block"
      >
        Does field contain personally identifying data?
      </span>
    </div>
  </div>
  <div
    className="col-xs-6"
    style={
      Object {
        "padding": "12px",
      }
    }
  >
    <div
      id="caldera-forms-privacy-gdpr-is-email-identifiying-fld_9899154"
    >
      Not an email or text field
    </div>
  </div>
</section>
`;

exports[`Settings Fields settings  Works 1`] = `
<div>
  <section
    className="row"
    style={
      Object {
        "backgroundColor": "#fff",
        "margin": "1em",
      }
    }
  >
    <h4
      style={
        Object {
          "borderBottom": "1px solid #eee",
          "margin": 0,
          "padding": ".7em 1em",
        }
      }
    >
      Priceaa
    </h4>
    <div
      className="col-xs-6"
      style={
        Object {
          "padding": "12px",
        }
      }
    >
      <div
        className="form-group"
      >
        <label
          className="control-label"
          htmlFor="caldera-forms-privacy-gdpr-is-pii-field-fld_9899154"
        >
          Personally Identifying Field?
        </label>
        <div
          className="checkbox"
        >
          <label
            title=""
          >
            <input
              checked={true}
              disabled={false}
              id="caldera-forms-privacy-gdpr-is-pii-field-fld_9899154"
              onChange={[Function]}
              type="checkbox"
            />
            <span
              style={
                Object {
                  "marginLeft": "12px",
                }
              }
            >
              Yes
            </span>
          </label>
        </div>
        <span
          className="screen-reader-text help-block"
        >
          Does field contain personally identifying data?
        </span>
      </div>
    </div>
    <div
      className="col-xs-6"
      style={
        Object {
          "padding": "12px",
        }
      }
    >
      <div
        id="caldera-forms-privacy-gdpr-is-email-identifiying-fld_9899154"
      >
        Not an email or text field
      </div>
    </div>
  </section>
  <section
    className="row"
    style={
      Object {
        "backgroundColor": "#fff",
        "margin": "1em",
      }
    }
  >
    <h4
      style={
        Object {
          "borderBottom": "1px solid #eee",
          "margin": 0,
          "padding": ".7em 1em",
        }
      }
    >
      Email
    </h4>
    <div
      className="col-xs-6"
      style={
        Object {
          "padding": "12px",
        }
      }
    >
      <div
        className="form-group"
      >
        <label
          className="control-label"
          htmlFor="caldera-forms-privacy-gdpr-is-pii-field-fld_4917648"
        >
          Personally Identifying Field?
        </label>
        <div
          className="checkbox"
        >
          <label
            title=""
          >
            <input
              checked={true}
              disabled={false}
              id="caldera-forms-privacy-gdpr-is-pii-field-fld_4917648"
              onChange={[Function]}
              type="checkbox"
            />
            <span
              style={
                Object {
                  "marginLeft": "12px",
                }
              }
            >
              Yes
            </span>
          </label>
        </div>
        <span
          className="screen-reader-text help-block"
        >
          Does field contain personally identifying data?
        </span>
      </div>
    </div>
    <div
      className="col-xs-6"
      style={
        Object {
          "padding": "12px",
        }
      }
    >
      <div
        className="form-group"
      >
        <label
          className="control-label"
          htmlFor="caldera-forms-privacy-gdpr-is-email-identifiying-fld_4917648"
        >
          Email Identifying Field?
        </label>
        <div
          className="checkbox"
        >
          <label
            title=""
          >
            <input
              checked={true}
              disabled={false}
              id="caldera-forms-privacy-gdpr-is-email-identifiying-fld_4917648"
              onChange={[Function]}
              type="checkbox"
            />
            <span
              style={
                Object {
                  "marginLeft": "12px",
                }
              }
            >
              Yes
            </span>
          </label>
        </div>
        <span
          className="screen-reader-text help-block"
        >
          Can this field be used to determine whose data an entry belongs to?
        </span>
      </div>
    </div>
  </section>
  <section
    className="row"
    style={
      Object {
        "backgroundColor": "#fff",
        "margin": "1em",
      }
    }
  >
    <h4
      style={
        Object {
          "borderBottom": "1px solid #eee",
          "margin": 0,
          "padding": ".7em 1em",
        }
      }
    >
      Total
    </h4>
    <div
      className="col-xs-6"
      style={
        Object {
          "padding": "12px",
        }
      }
    >
      <div
        className="form-group"
      >
        <label
          className="control-label"
          htmlFor="caldera-forms-privacy-gdpr-is-pii-field-fld_1081036"
        >
          Personally Identifying Field?
        </label>
        <div
          className="checkbox"
        >
          <label
            title=""
          >
            <input
              checked={false}
              disabled={false}
              id="caldera-forms-privacy-gdpr-is-pii-field-fld_1081036"
              onChange={[Function]}
              type="checkbox"
            />
            <span
              style={
                Object {
                  "marginLeft": "12px",
                }
              }
            >
              Yes
            </span>
          </label>
        </div>
        <span
          className="screen-reader-text help-block"
        >
          Does field contain personally identifying data?
        </span>
      </div>
    </div>
    <div
      className="col-xs-6"
      style={
        Object {
          "padding": "12px",
        }
      }
    >
      <div
        id="caldera-forms-privacy-gdpr-is-email-identifiying-fld_1081036"
      >
        Not an email or text field
      </div>
    </div>
  </section>
</div>
`;
